src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.cs

```diff
@@ -132,6 +133,36 @@ protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pin
         /// <inheritdoc/>
         protected internal override void Write(int pinNumber, PinValue value) => _internalDriver.Write(pinNumber, value);
 
+        /// <summary>
+        /// Returns the currently set pin mode by directly reading the hardware
+        /// </summary>
+        /// <param name="pinNumber">Pin number</param>
+        /// <returns>(Alternate) Pin mode. 0 = Alt0, 1= Alt1... -1 Gpio Input, -2 Gpio Output</returns>
+        protected internal int GetAlternatePinMode(int pinNumber)

```

Comment:
should this be an enum?

Reply:
Ah, this is about current pin mode, not all possible options...

Reply:
I would really like to use an enum here. Unfortunately, the set of alternate pin modes is hardware dependent. So if I go with the names that are most common here, they would be Raspberry Pi-specific and be confusing to use on some other hardware (the Arduino binding for instance uses a similar semantics, but with different values)

Maybe return a class instead? (That would be handled much like a handle for most of what the client does)?

Reply:
if you have some decent design with struct (class is too heavy to be used instead of enum) then fine, otherwise I'm fine with leaving as is

Reply:
Used a class with static members now. This syntactically looks identical to using an enum, but is extensible.